CHANGELOG.md

```diff
@@ -3,6 +3,8 @@
 **Unreleased**
 --------------
 
+- **Enhancement**: Report more context for error types, such as the name of the parameter it came from.
```

Comment:
Also, it checks error types in the generic arguments, which I think was the main pain point.

Reply:
```suggestion
- **Enhancement**: Report more context for error types, such as the name of the parameter it came from.
- **Fix**: For generic types, check for error types in their type arguments as well.
```